### DISCUSSION TITLE
Whether it is allowed to overwrite get_context?

### DISCUSSION THREAD

**Discussion by Matieus** [Rapptz/discord.py#9679] [Q&A] [answered]
Can I replace the get_prefix method in my own way in custom bot class? My idea is to download a set of prefixes and then update them whenever there is a change in the file and the variable.

I want to do this to have the fastest access to the data. For example, the dict type allows this because it uses hashes.
My idea is to download prefixes from the database or json file at the start of the bot.
Is it safe?

    from typing import Any

    from discord import Interaction, Message
    from discord.ext import commands

    from .context import CustomContext

    class MyCustomBot(commands.Bot):
        """A subclass of commands.Bot, useful for creating custom context."""
        prefixes = {}

        async def get_context(
            self, message: Message | Interaction, *, cls: Any = CustomContext
        ) -> CustomContext:
            return await super().get_context(message, cls=cls)

        async def get_prefix(self, message: Message) -> str:
            if message.guild:
                return self.prefixes.get(str(message.guild.id), "$")
            else:
                return "!"

**Comment by LostLuma** [comment, marked as answer]
You can do this and it is indeed the intended method of doing so, we even have an [entire example](https://github.com/Rapptz/discord.py/blob/master/examples/custom_context.py) explaining it for custom Contexts.

**Reply by Matieus** [reply]
Thx

**Comment by Soheab** [comment]
Although what you want is possible but you don't need to override `get_prefix` for it. Instead the [`command_prefix`](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html#discord.ext.commands.Bot.command_prefix) argument can take a callable that is called with your `bot` and the `message`:
```py
async def custom_prefix(bot: commands.Bot, message: discord.Message):
    ...
    return "!"


commands.Bot(command_prefix=custom_prefix, ...)
...
```

**Comment by Matieus** [comment]
That is all, thank you
